### DISCUSSION TITLE
Passing string value for custom profiler to the trainer.

### DISCUSSION THREAD

**Discussion by ankitgola005** [Lightning-AI/pytorch-lightning#16563] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I have been working on a custom profiler installable package for which I want to pass string option to the trainer. However trainer has a check in _init_profiler that allows string values to only be one of "simple", "advanced", "pytorch", or "xla". 
Is there a way to do this in an external package without modifying the trainer in PyTorch Lightning ?

Error: raise MisconfigurationException(
lightning_lite.utilities.exceptions.MisconfigurationException: When passing string value for the `profiler` parameter of `Trainer`, it can only be one of ['simple', 'advanced', 'pytorch', 'xla']

**Comment by tchaton** [comment, marked as answer]
Dear @assassin1991,

Unfortunately, PyTorch Lightning doesn't support passing new profiler strings as those are already registered in the framework.  

Here are several options:
1) Pass the profiler directly to the Trainer. Simpler solution IMO especially if you are exposing arguments to the profiler.
2) Patch the Trainer internal to register your profiler string but this might break if internal changes
3) Open a PR enabling users to register new profiler names. If your profiler subclasses PyTorch Lightning profilers and register a static `name`, I believe it is possible for the Trainer to search all the available subclass of the BaseProfiler and perform automatic resolution of the Profiler class.

Best,
T.C

**Reply by ankitgola005** [reply]
@tchaton . Thank you so much. I think I will continue with option1 for now.
